sympy/parsing/tests/test_sympy_parser.py

```diff
@@ -168,6 +168,21 @@ def test_recursive_evaluate_false_10560():
         assert parse_expr(text, evaluate=False) == parse_expr(result, evaluate=False)
 
 
+def test_function_evaluate_false():
+    inputs = [
+        'Abs(0)', 'im(0)', 're(0)', 'sign(0)', 'arg(0)', 'conjugate(0)',
+        'acos(0)', 'acot(0)', 'acsc(0)', 'asec(0)', 'asin(0)', 'atan(0)',
+        'acosh(0)', 'acoth(0)', 'acsch(0)', 'asech(0)', 'asinh(0)', 'atanh(0)',
+        'cos(0)', 'cot(0)', 'csc(0)', 'sec(0)', 'sin(0)', 'tan(0)',
+        'cosh(0)', 'coth(0)', 'csch(0)', 'sech(0)', 'sinh(0)', 'tanh(0)',
+        'exp(0)', 'ln(0)', 'log(0)',
+    ]
+    for case in inputs:
+        expr = parse_expr(case, evaluate=False)
+        assert isinstance(expr, Function)
+        assert str(expr.doit()) != case  # in case function name is wrongly typed
```

Comment:
I had to read the comment a few time before getting that "in case" didn't refer to `case`...

Reply:
Perhaps
```
assert case == str(expr) != str(expr.doit())
```

Reply:
> Perhaps
> 
> ```
> assert case == str(expr) != str(expr.doit())
> ```

not work for ln as it will be converted to log

Reply:
Maybe test `ln` separately then. Ideally that kind of difference in behaviour should be visible in the test code.

Reply:
> Maybe test `ln` separately then. Ideally that kind of difference in behaviour should be visible in the test code.

done